### ISSUE TITLE
[Bug]: Images still not loading with shopify theme dev on 3.46.3

### ISSUE THREAD

**Issue by MattWIP** [Shopify/cli#2132]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

When running `shopify theme dev` the images rendered using `image_url` should provide a valid url from the local IP that retreives the image from Shopify.

This issue seems to be patched by 3.46.3 but even after the upgrade the issue has persisted for me.

### Actual behavior

When running `shopify theme dev` the images rendered using `image_url` return a url that does not render a valid url. If we use `img_url` the returned URL works because its pointing to the stores actual domain instead of our local IP.

The image_url filter produces an image url like this:
http://[my-ip]:9292/cdn/shop/files/image.jpg?crop=center&amp;v=1686004089&amp;width=1678

which points to a non-rendering image.

### Verbose output

```shell
[0] DEBUG [BulkJob #2040] job response: http_status=207
[0] DEBUG [BulkJob #2040] bulk item error (item=package.json, status=404, retries=8)
[0] DEBUG [BulkJob #2140] job request: size=1, bytesize=2495
[0] DEBUG PUT https://[store].myshopify.com/admin/api/unstable/themes/132087578784/assets/bulk.json with request_id: a3a60cb3-ddd6-46c5-965c-f2061699f958
[0] DEBUG [BulkJob #2140] bulk rest_request time: 280.4505ms
[0] DEBUG [BulkJob #2140] job response: http_status=207
[0] DEBUG [BulkJob #2140] bulk item error (item=package.json, status=404, retries=9)
[0] DEBUG [BulkJob #2120] job request: size=1, bytesize=2495
[0] DEBUG PUT https://[store].myshopify.com/admin/api/unstable/themes/132087578784/assets/bulk.json with request_id: 3d3acf5e-7127-49cb-93d7-22917c8f6e52
[0] DEBUG [BulkJob #2120] bulk rest_request time: 217.3112ms
[0] DEBUG [BulkJob #2120] job response: http_status=207
[0] DEBUG [BulkJob #2120] bulk item fatal error (item=package.json, status=404)
[0] DEBUG [Bulk] shutdown, remaining_items=0                                    100%
[0] package.json: package.json is not a theme file
[0]                                                                             100%
[0] Please open this URL in your browser:
[0] http://[ip]:9292
[0] [2023-06-08 08:37:45] INFO  WEBrick 1.7.0
[0] [2023-06-08 08:37:45] INFO  ruby 3.0.2 (2021-07-07) [x64-mingw32]
[0] [2023-06-08 08:37:45] INFO  WEBrick::HTTPServer#start: pid=30832 port=9292
[0] DEBUG Refreshing preview _secure_session_id cookie
[0] DEBUG Proxying HEAD https://[store].myshopify.com/?preview_theme_id=132087578784&_fd=0&pb=0
[0] DEBUG `-> 302 request_id: fc041287-136a-4440-a0fe-7d498e9a8338
[0] DEBUG Proxying GET https://[store].myshopify.com/?_fd=0&pb=0
[0] DEBUG `-> 200 request_id: 88452b0d-9242-4971-a380-6e0524dda8b6
[0] DEBUG Proxying GET https://[store].myshopify.com/?_fd=0&pb=0
[0] DEBUG `-> 200 request_id: 2e64aa75-3ae1-47f9-9104-8e22bcf1b208
[1] 
[1] warn - The utility `w-[calc(100%-theme('spacing[some_key][1.5]'))]` contains an invalid theme value and was not generated.
[0] DEBUG Proxying GET https://[store].myshopify.com/cart.js?_fd=0&pb=0
[0] DEBUG [HotReload] Connected to SSE stream
[0] DEBUG `-> 200 request_id: c1e3aee9-e699-41d6-858a-7d004dc2653d
[0] DEBUG Proxying GET https://[store].myshopify.com/cart.js?_fd=0&pb=0
[0] DEBUG `-> 200 request_id: ec3a2f30-3c94-4488-a790-6cd0ef541875
[0] DEBUG Proxying GET https://[store].myshopify.com/products/routeins.js?_fd=0&pb=0
[0] DEBUG Proxying GET https://[store].myshopify.com/cart.js?timestamp=1686238687409&_fd=0&pb=0
[0] DEBUG `-> 404 request_id: 545bc231-d8bd-4d1c-8a4f-ac5484cf4768
[0] DEBUG `-> 200 request_id: c0475fc8-58e9-463a-bfa9-fc501918099a
```

### Reproduction steps

1. run `shopify theme dev`
2. render an image using `{{ image | image_url: width: 500 }}`
3. See it not load in.

### Operating System

Windows 10

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.46.3

**Comment by amcaplan** [comment]
cc @Shopify/theme-code-tools

**Comment by karreiro** [comment]
Thank you for reporting this, @MattWIP!

It seems like the same issue reported by https://github.com/Shopify/cli/issues/2120. If, so I'm working on the fix for this. 

Still, could you please follow the steps below, so I can be sure it's the same issue that you're going to have a fix in the next CLI release:

1. Re-run the command, but this time include the `--verbose` flag.
2. Capture a screenshot that displays both the terminal and network logs, similar to the example provided below:

    ![Screenshot 2023-06-08 at 16 10 48](https://github.com/Shopify/cli/assets/1079279/4f227a85-5ae5-4c2e-a92e-da2dc4c19f0d)


Thank you!

**Comment by karreiro** [comment]
Thank you for providing those request IDs, @MattWIP. They were helpful in validating the fix for [Shopify/cli#2138](https://github.com/Shopify/cli/pull/2138).

If you update the CLI to version 3.46.5, you should no longer encounter this issue.
